From 032244109f0404db161844f971908e857c8e591c Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sat, 09 Jul 2022 17:53:28 +0300
Subject: [PATCH] [SW-94367] gaudi3: clear boot termination even if link is up

Currenlty, clearing of boot terimnation is performed as part
of PCIe link up routine. However there is possibility if
boot termination is enabled somewhere in flow or as part of reset
sequence. To avoid such scenario, this patch adds clearing
of boot termination even if PCIe link is detected in ROM recovery
flow.

Change-Id: I6c896229a9b33afe0f8fa0e28201f0f09b1033ec
---

diff --git a/common/pcie.c b/common/pcie.c
index 74b10be..6748c8b 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -148,6 +148,8 @@
 			      mmCPU_HOT_RST);
 
 		regmap_writel(0x3, mmLTSSM_EN);
+	} else {
+		pcie_config_boot_termination();
 	}
 
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH);
